vowpalwabbit/conditional_contextual_bandit.cc

```diff
@@ -0,0 +1,300 @@
+#include "conditional_contextual_bandit.h"
+
+#include "example.h"
+#include "global_data.h"
+
+#include <numeric>
+
+bool CCB::ec_is_example_header(example& ec)
+{
+  return ec.l.conditional_contextual_bandit.type == example_type::shared;
+}
+
+void CCB::print_update(vw& all, bool is_test, example& ec, multi_ex* ec_seq, bool action_scores)
+{
+  // TODO: Implement for CCB

```

Comment:
This is the exponential backoff printout which VW does by default, so it's obviously important to implement.

Reply:
Yes.   Early tomorrow morning?